### PR TITLE
fix: rebase error message during conflicts

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Updated the error message for rebase during the conflicts state

Fixes https://github.com/appsmithorg/appsmith/issues/24391

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

## Testing
- [ ] Manual
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by nayan-rafiq — DISMISSED** [review]

**Review by nayan-rafiq — DISMISSED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -2213,15 +2213,15 @@ public Mono<Application> discardChanges(String defaultApplicationId, String bran
...
-                                    return Mono.error(new AppsmithException(AppsmithError.GIT_ACTION_FAILED, "rebase branch", "Unable to rebase branch - " + branchName));
-                                }
                                 return fileUtils.reconstructApplicationJsonFromGitRepo(
                                         branchedApplication.getWorkspaceId(),
                                         branchedApplication.getGitApplicationMetadata().getDefaultApplicationId(),
                                         branchedApplication.getGitApplicationMetadata().getRepoName(),
                                         branchName);
                             })
+                            .onErrorResume(throwable -> {
+                                return Mono.error(new AppsmithException(AppsmithError.GIT_ACTION_FAILED, "discard changes", "Please create a new branch and resolve the conflicts on remote repository before proceeding ahead."));
```
Comment (nayan-rafiq) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:2223]: We should log the exception as well here.

Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:2223]: This is logged in already `GitExecutorImpl` class

**Review by Parthvi12 — APPROVED** [review]

**Review by Parthvi12 — APPROVED** [review]

**Review by brayn003 — APPROVED** [review]

**Pull request merged**